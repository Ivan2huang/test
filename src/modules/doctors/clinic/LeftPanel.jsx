/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import { Box, Link, Hidden, withStyles } from '@material-ui/core';

import { VIEW } from './constants';
import ClinicListViewContainer from './ClinicListViewContainer';
import ClinicDetailViewContainer from './ClinicDetailViewContainer';
import ClinicSearchContainer from './ClinicSearchContainer';
import ClinicFilterContainer from './ClinicFilterContainer';
import { formatMessage } from '../../../helpers/helpers';
import Typography from '../../../uiComponents/Typography';

const styles = theme => ({
  sideBar: {
    minWidth: '100%',
    maxHeight: '100%',
    boxShadow: `0px 1px 0px ${theme.grey1}`,
    [theme.breakpoints.up('md')]: {
      width: '30%',
      minWidth: theme.spacingX(73),
    },
  },
  searchField: {
    boxShadow: `0 1px 0 ${theme.grey1}`,
    marginBottom: '1px',
    background: theme.white,
    paddingLeft: theme.spacingX(8),
    paddingRight: theme.spacingX(8),
    paddingTop: theme.spacingX(6),
    paddingBottom: theme.spacingX(6),
    [theme.breakpoints.down('xs')]: {
      paddingLeft: '0px',
      paddingRight: '0px',
      background: theme.transparent,
    },
  },
  listViewContainer: {
    background: theme.background,
    display: 'flex',
    flex: 1,
    overflowY: 'auto',
    justifyContent: 'center',

    [theme.breakpoints.down('xs')]: {
      marginLeft: `-${theme.spacingX(8)}`,
      marginRight: `-${theme.spacingX(8)}`,
    },
  },
  filterWrapper: {
    [theme.breakpoints.down('sm')]: {
      display: 'flex',

      '& > div:first-child': {
        flex: 1,
      },

      '& > div:last-child': {
        marginLeft: theme.spacingX(5),
      },
    },
  },
});

const LeftPanel = ({ intl, classes, view, toggleView }) => {
  const nextView = () => (view === VIEW.list ? VIEW.map : VIEW.list);
  const [isSearchOpen, setIsSearchOpen] = useState(false);
  const [isFilterOpen, setIsFilterOpen] = useState(false);

  return (
    <Box
      display="flex"
      flexDirection="column"
      className={classes.sideBar}
      height={view === VIEW.list ? '100%' : 'auto'}
    >
      <Box className={classes.searchField}>
        <ClinicSearchContainer open={isSearchOpen} setOpen={setIsSearchOpen} />
        <div className={classes.filterWrapper}>
          <ClinicFilterContainer
            open={isFilterOpen}
            setOpen={setIsFilterOpen}
            closeSearchPopup={() => setIsSearchOpen(false)}
          />
          <Box
            mt={5}
            display="flex"
            justifyContent="flex-end"
            implementation="css"
            mdUp
            component={Hidden}
          >
            <Link
              data-testid="btn-toggle-view"
              underline="none"
              component="button"
              onClick={toggleView}
            >
              <Typography type="style-6" color="hyperlink">
                {formatMessage(
                  intl,
                  'clinic.searchClinics.toggleView',
                  `View ${nextView()}`,
                  {
                    view: nextView(),
                  },
                )}
              </Typography>
            </Link>
          </Box>
        </div>
      </Box>
      <Box className={classes.listViewContainer}>
        <ClinicListViewContainer
          closeFilterPopup={() => setIsFilterOpen(false)}
        />

        <ClinicDetailViewContainer />
      </Box>
    </Box>
  );
};

LeftPanel.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({
    sideBar: PropTypes.string.isRequired,
    searchField: PropTypes.string.isRequired,
    listViewContainer: PropTypes.string.isRequired,
    filterWrapper: PropTypes.string.isRequired,
  }).isRequired,
  view: PropTypes.string.isRequired,
  toggleView: PropTypes.func.isRequired,
};

export default injectIntl(withStyles(styles)(LeftPanel));
