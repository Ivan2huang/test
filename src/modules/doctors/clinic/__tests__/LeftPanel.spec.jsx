import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LeftPanel from '../LeftPanel';
import withIntl from '../../../../i18n/withIntlProvider';
import withTheme from '../../../../themes/withThemeProvider';

jest.mock('../ClinicListViewContainer', () => () => (
  <div>Clinic List View Container</div>
));
jest.mock('../ClinicDetailViewContainer', () => () => (
  <div>Clinic Details View Container</div>
));
jest.mock('../ClinicSearchContainer', () => () => (
  <div>Clinic search Container</div>
));
jest.mock('../ClinicFilterContainer', () => () => (
  <div>Clinic Filter Container</div>
));

describe('LeftPanel Component', () => {
  const defaultProps = {
    view: 'map',
    toggleView: jest.fn(),
  };
  const Component = withTheme(withIntl(LeftPanel));

  it('should match the snapshot', () => {
    const { container } = render(<Component {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should be able to toggle the view  when change view clicked', () => {
    const { container, getByTestId, rerender } = render(
      <Component {...defaultProps} />,
    );

    const changeView = getByTestId('btn-toggle-view');
    fireEvent.click(changeView);
    expect(defaultProps.toggleView).toHaveBeenCalledTimes(1);
    const newProps = {
      ...defaultProps,
      view: 'list',
    };
    rerender(<Component {...newProps} />);
    expect(container).toMatchSnapshot();
  });
});
