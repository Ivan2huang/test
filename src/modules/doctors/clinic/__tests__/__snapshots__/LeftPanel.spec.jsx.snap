// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftPanel Component should be able to toggle the view  when change view clicked 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-68 LeftPanel-sideBar-63"
  >
    <div
      class="MuiBox-root MuiBox-root-69 LeftPanel-searchField-64"
    >
      <div>
        Clinic search Container
      </div>
      <div
        class="LeftPanel-filterWrapper-66"
      >
        <div>
          Clinic Filter Container
        </div>
        <div
          class="MuiBox-root MuiBox-root-70 PrivateHiddenCss-mdUp-78"
        >
          <button
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiLink-button MuiTypography-colorPrimary"
            data-testid="btn-toggle-view"
          >
            <div
              class="MuiTypography-root makeStyles-root-125 makeStyles-root-126 MuiTypography-body1"
            >
              View map
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-124 LeftPanel-listViewContainer-65"
    >
      <div>
        Clinic List View Container
      </div>
      <div>
        Clinic Details View Container
      </div>
    </div>
  </div>
</div>
`;

exports[`LeftPanel Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6 LeftPanel-sideBar-1"
  >
    <div
      class="MuiBox-root MuiBox-root-7 LeftPanel-searchField-2"
    >
      <div>
        Clinic search Container
      </div>
      <div
        class="LeftPanel-filterWrapper-4"
      >
        <div>
          Clinic Filter Container
        </div>
        <div
          class="MuiBox-root MuiBox-root-8 PrivateHiddenCss-mdUp-16"
        >
          <button
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiLink-button MuiTypography-colorPrimary"
            data-testid="btn-toggle-view"
          >
            <div
              class="MuiTypography-root makeStyles-root-60 makeStyles-root-61 MuiTypography-body1"
            >
              View list
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-62 LeftPanel-listViewContainer-3"
    >
      <div>
        Clinic List View Container
      </div>
      <div>
        Clinic Details View Container
      </div>
    </div>
  </div>
</div>
`;
